packages/backend-api/src/Controller/V1/Product/ProductApiDataValidator.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\BackendApiBundle\Controller\V1\Product;
+
+use DateTime;
+use Shopsys\BackendApiBundle\Component\Validation\ValidationRunner;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Symfony\Component\Validator\Constraints;
+
+/**
+ * @experimental
+ */
+class ProductApiDataValidator implements ProductApiDataValidatorInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Shopsys\BackendApiBundle\Component\Validation\ValidationRunner
+     */
+    protected $validationRunner;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\BackendApiBundle\Component\Validation\ValidationRunner $validationRunner
+     */
+    public function __construct(Domain $domain, ValidationRunner $validationRunner)
+    {
+        $this->domain = $domain;
+        $this->validationRunner = $validationRunner;
+    }
+
+    /**
+     * @param array $productApiData
+     * @return string[]
+     */
+    public function validateCreate(array $productApiData): array
+    {
+        $constraintCollection = new Constraints\Collection($this->getConstraintDefinition());
+
+        return $this->validationRunner->runValidation($productApiData, $constraintCollection);
+    }
+
+    /**
+     * @param array $productApiData
+     * @return string[]
+     */
+    public function validateUpdate(array $productApiData): array
+    {
+        $constraintCollection = new Constraints\Collection($this->getConstraintDefinition());
+
+        return $this->validationRunner->runValidation($productApiData, $constraintCollection);
```

Comment:
As this method should behave the same as `validateCreate` (according to documentation), I prefer to simplify it to just 
```php
return $this->validateCreate($productApiData);
```